operator/v1/types_console.go

```diff
@@ -143,8 +143,94 @@ type Capability struct {
+	// +kubebuilder:validation:Enum="Dark";"Light";"Default"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// file is used by the console to locate the specified file containing a custom logo.
+	// file is a required field that references a ConfigMap name and key that contains the custom logo file in the openshift-config namespace.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file'
+	// The ConfigMap key must include the file extension for the dark, light, or default theme so that the console serves the file with the correct MIME type, which can be '.svg', '.png', or whatever the browser supports.
+	// Recommended Masthead and About Modal logo specifications:
+	// Image size must be less than 1 MiB due to constraints on the ConfigMap size.
+	// Dimensions: Max height of 60px and max width of 200px.
+	// SVG format preferred.
+	// Recommended Favicon specifications:
+	// Dimensions: Max height of 16px and max width of 16px
+	// PNG format preferred.
+	// +required
+	File configv1.ConfigMapFileReference `json:"file"`
+}
+
+// CustomLogo defines a configuration based on theme types for the console UI logo.
+type CustomLogo struct {
+	// type specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+	// type is a required field that allows values of Masthead and Favicon.
+	// When set to "Masthead", the logo will be used in the masthead and about modal of the console UI.
+	// When set to "Favicon", the logo will be used as the favicon of the console UI.
+	// +kubebuilder:validation:Enum="Masthead";"Favicon"
+	// +required
+	Type LogoType `json:"type"`
+
+	// themes specifies the themes for the console UI logo.
+	// themes is a required field that allows a list of themes. Each item in the list must have a unique Type and a File field.
+	// Each theme determines whether the logo is for the dark or light or default theme of the console UI.
+	// There must be at least one entry and no more than 3 entries.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=3
+	// +listType=map
+	// +listMapKey=type
+	// +required
+	Themes []Theme `json:"themes"`
+}
+
 // ConsoleCustomization defines a list of optional configuration for the console UI.
 type ConsoleCustomization struct {
+	// customLogos is used to replace the OpenShift Masthead and Favicon logos in the console UI with custom logos.
+	// customLogos is an optional field that allows a list of custom logos.
+	// When specified, there must be at least one entry and no more than 2 entries.
+	// Each item in the list must have a unique Type and a Themes field.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	CustomLogos []CustomLogo `json:"customLogos,omitempty"`
```

Comment:
Since this is a configuration API, we may want to look at not having the `omitempty` and the `minItems:=1` for discoverability purposes

Reply:
`omitempty` and the `minItems:=1` makes sense to me

Reply:
My suggestion was to remove the omitempty.

IIUC, odds are that this resource is created by something that marshals down from Go, so we can force

```
customLogso: []
```

to appear. This would then prod the user to understand that they have an option to configure the logos if they ever fetch the resource. We call this config discoverability.

Reply:
Sorry @JoelSpeed, I misspoke. I meant I agree with removing those props as you suggest